src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/MetadataTests/DefaultJsonTypeInfoResolverTests.JsonPropertyInfo.cs

```diff
@@ -0,0 +1,1048 @@
+                                return intVal != 3;
+                            }
+                            else if (v is string stringVal)
+                            {
+                                return stringVal != "asd";
+                            }
+
+                            Assert.Fail("ShouldSerialize set for value which is not int or string");
+                            return false;
+                        };
+                        break;
+                }
+            }
+        }
+
+        private class TestClassWithEveryPossibleJsonIgnore
+        {
+            [JsonIgnore(Condition = JsonIgnoreCondition.Always)]
+            public string AlwaysProperty { get; set; }
+
+            [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
+            public int WhenWritingDefaultProperty { get; set; }
+
+            [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+            public string WhenWritingNullProperty { get; set; }
+
+            [JsonIgnore(Condition = JsonIgnoreCondition.Never)]
+            public string NeverProperty { get; set; }
+
+            public string Property { get; set; }
+        }
+
+        private class TestClassWithCustomConverterOnProperty
+        {
+            [JsonConverter(typeof(MyClassConverterOriginal))]
+            public MyClass MyClassProperty { get; set; }
+        }
+
+        private class MyClassConverterOriginal : JsonConverter<MyClass>
+        {
+            public override MyClass? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+            {
+                if (reader.TokenType != JsonTokenType.String)
+                    throw new InvalidOperationException($"Wrong token type: {reader.TokenType}");
+
+                MyClass myClass = new MyClass();
+                myClass.Value = reader.GetString();
+                return myClass;
+            }
+
+            public override void Write(Utf8JsonWriter writer, MyClass value, JsonSerializerOptions options)
+            {
+                writer.WriteStringValue(value.Value);
+            }
+        }
+
+        private class MyClassCustomConverter : JsonConverter<MyClass>
+        {
+            private string _prefix;
+
+            public MyClassCustomConverter(string prefix)
+            {
+                _prefix = prefix;
+            }
+
+            public override MyClass? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+            {
+                if (reader.TokenType != JsonTokenType.String)
+                    throw new InvalidOperationException($"Wrong token type: {reader.TokenType}");
+
+                MyClass myClass = new MyClass();
+                myClass.Value = reader.GetString().Substring(_prefix.Length);
+                return myClass;
+            }
+
+            public override void Write(Utf8JsonWriter writer, MyClass value, JsonSerializerOptions options)
+            {
```

Comment:
Do we have JsonPropertyInfo unit tests for the following scenaria:

* Creating a JsonPropertyInfo from `JsonTypeInfo<Foo>` and then trying to add it to the list of properties in `JsonTypeInfo<Bar>`. Should include scenaria in which `Foo` and `Bar` are in a type relationship or the metadata are different instances of the same type.
* The `CustomConverter` should return the `JsonConverterFactory` instance and not the fully expanded converter.
* Unit tests validating the mapping between `JsonIgnoreCondition` and `ShouldSerialize`.
* Validating behavior when assigning getters and setters that use invalid types.

Reply:
This is partially address. Remaining work:
- validate assigning ShouldSerialize = null for different default JsonIgnoreCondition (there is some testing there but not exhaustive)
- Validating behavior when assigning getters and setters that use invalid types.